src/qibolab/backends.py

```diff
@@ -50,6 +52,50 @@ def __init__(self, platform):
         }
         self.compiler = Compiler.default()
 
+    @property
+    def qubits(self) -> list[Union[str, int]]:
+        return list(self.platform.qubits)
+
+    @property
+    def connectivity(self) -> list[tuple[Union[str, int], Union[str, int]]]:
+        return list(self.platform.pairs)
+
+    @property
+    def natives(self) -> list[str]:
+        native_gates = set()
+        for _, q in self.platform.qubits.items():
+            native_gates |= {
+                fld.name
+                for fld in fields(q.native_gates)
+                if getattr(q.native_gates, fld.name) is not None
+            }
+
+        for _, p in self.platform.pairs.items():
+            native_gates |= {
+                fld.name
+                for fld in fields(p.native_gates)
+                if getattr(p.native_gates, fld.name) is not None
+            }
+
+        return self._rename_gates(list(native_gates))
+
+    def _rename_gates(self, natives: list[str]):
+        gates = []
+        for native in natives:
+            if native == "RX12":
+                pass
+            elif native == "RX":
+                gates.append("RX")
+            elif native == "CZ":
+                gates.append("CZ")
+            elif native == "iSWAP":
+                gates.append("iSWAP")
+            elif native == "MZ":
+                gates.append("M")
+            elif native == "CNOT":
+                gates.append("CNOT")
+        return gates
```

Comment:
It's not just a matter of renaming gates: as @stavros11 noted above, the gates you're interested in are those defined by the compiler, not the calibrated ones, which you can not directly access.

Thus, you should not directly inspect the `self.platform.qubits[q].native_gates` attribute, but rather you should take them from here:
https://github.com/qiboteam/qibolab/blob/32187d4c349544a757ee6f9d74c07e67bc88e7a5/src/qibolab/compilers/compiler.py#L45-L59
filtering out those for which you do not have an associated calibrated pulse (you can see that they internally [call a platform method](https://github.com/qiboteam/qibolab/blob/32187d4c349544a757ee6f9d74c07e67bc88e7a5/src/qibolab/compilers/default.py#L82-L93) which may fail if the calibration is absent).

Reply:
Notice that, as pointed out by @stavros11, there is no `RX` gate, but rather `GPI2`/`GPI`.

Reply:
Ok. In principle, we should check every qubit and edge to see if they have rules. But since Qibo does not support that specific compilation now, should we just return all of them?

Reply:
Indeed, we can not transpile with per-qubit natives, so it's useless to return them (that's why I did not propose it).

However, I would actually return all of them, but filter the `CZ` and `CNOT` based on a single qubit pair (just take the first, and check what's in there). There are some chips that should calibrate the `CZ` (and possibly the `iSWAP`, but that's not yet in the compiler...), and others that will calibrate the `CNOT`.
Ttbomu, it's either one or the other, not both - so returning both would make it impossible to correctly any circuit that contains both of them (or decompose to both). So, they have to be filtered to achieve minimal usability.

Reply:
> Is there a better way to avoid using dunders? We need `__name__` and `__class__` to handle `gates`. I think we should implement encapsulation in `qibo.gates`.

Ah, sorry, I missed your reply, and suggested the same below :P

Yes, you're definitely right. However, the type is Python-wide feature, and the `type()` constructor can be used exactly for that purpose.

`__name__` is a different story, since we're actually using it as a string in various places, but this is just our own convention. So, yes, it should be encapsulated where the convention is established.

However, this is happening because mainly because of the transpiler. In particular, because of the unroller in:
https://github.com/qiboteam/qibo/blob/45eae7f81d1fe0fe9c3cd5b39c98fa225a6c29e8/src/qibo/transpiler/unroller.py#L47-L55

So, it is not strictly required to move this to the gates, since we could do in the place where we're defining the strings (i.e. the `NativeGates` class).
But it's true that this is not the only place, so we could even do that in the gates themselves.